src/AbstractValidator.php

```diff
@@ -796,8 +796,8 @@ private function checkFileTypeWithFinfo(string $name, array $types): bool
     // type from there.  this should mean that even files from Macs,
     // i.e. without extensions, should be identifiable.
     
-    $info = new finfo(FILEINFO_MIME_TYPE);
-    $type = $info->file($_FILES[$name]["tmp_name"]);
+    $info = finfo_open(FILEINFO_MIME_TYPE);
+    $type = finfo_file($info, $_FILES[$name]["tmp_name"]);
     
     if ($type === false) {
```

Comment:
for a reason I don't understand, using the file info extension in OOP mode confused my IDE; a warning was flagged on the `$type === false` comparison despite the PHP docs clearly indicating that `finfo::file` returns `string|bool` just like the `finfo_file` does.  but, it made my IDE happy to switch from OOP to functional calls and it all works the same way.